src/test/java/testHW9/FlightBookingTest.java

```diff
@@ -0,0 +1,76 @@
+package testHW9;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.PageFactory;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class FlightBookingTest extends BaseTest {
+    private BaseTestAdmin baseTestAdmin;
+    //Create test data
+    String departureCity ="POS";
+    String destinationCity = "MYF";
+    String flightClass = "business";
+    int adultsNum = 2;
+    int childrenNum = 1;
+    int infantsNum = 1;
+
+    String firstName = "Oksana";
+    String lastName = "Skovron";
+    String email = "os@gmail.com";
+    String contactNumber = "3547877";
+    String address = "address 1 /7";
+    String country = "Spain";
+    String passengersName = "Passenger1";
+    String passengersAge = "25";
+    String passportNo = "ES58744";
+
+    @Test
+    public void testBooking(){
+        //Go to Flights tab, enter data and click Search
+        homePage.searchFlights(departureCity, destinationCity, flightClass, adultsNum, childrenNum, infantsNum);
```

Comment:
You're initializing pageobjects the wrong way.
Instead of initializing all of them in BaseTest - you should do the following instead:
`searchFlightResultsPage = homePage.searchFlights(departureCity, destinationCity, flightClass, adultsNum, childrenNum, infantsNum);`
And then:
`bookingPage = searchFlightResultsPage.clickBookNowButton();`
And so on and so forth.